File: Problem Set 2/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>

int main(int argc, string argv[])
{
    bool isValidKey(int argc, string key);
    string encryptText(string plaintext, int key);
    
    // Quit the program if exactly 1 argument was not provided
    if (argc != 2)
    {
        printf("Usage: ./caesar key\n");
        return 1;
    } 
        
    if (!isValidKey(argc, argv[1]))
    {
        printf("Usage: ./caesar key\n");
        return 1;
    }
    
    // Convert the key to an integer
    int validKey = atoi(argv[1]);
    
    string plaintext = get_string("plaintext: ");
    string ciphertext = encryptText(plaintext, validKey);
    
    printf("ciphertext: %s\n", ciphertext);
    
    return 0; 
}

// Checks whether the key provided in the command is valid
bool isValidKey(int argc, string key) 
{
    for (int i = 0; i < strlen(key); i++)
    {
        if (!isdigit(key[i]))
        {
            return false;
        }
    }
    
    return true;
}

// Encrypts the plaintext provided by the user
string encryptText(string plaintext, int key)
{
    for (int i = 0; i < strlen(plaintext); i++)
    {       
        if (islower(plaintext[i]))
        {
            plaintext[i] = 97 + (plaintext[i] + key - 97) % 26;
        } 
        else if (isupper(plaintext[i]))
        {
            plaintext[i] = 65 + (plaintext[i] + key - 65) % 26;
        }
    }
    
    return plaintext;
}

File: Problem Set 6/caesar/caesar.py
from cs50 import get_string
from sys import argv


def main():
    # If the user doesn't provide exactly one argument, print usage and exit
    while len(argv) != 2:
        print("Usage: python caesar.py k")
        exit(1)
    
    # Convert the key to an integer
    validKey = int(argv[1])
   
    # Get the plaintext from the user
    plaintext = get_string("plaintext: ")
    
    # Use a function to encrypt the plaintext
    ciphertext = encryptText(plaintext, validKey)
    
    # Return the encrypted text
    print("ciphertext: {c}".format(c=ciphertext))


# Function which encrypts a given text based on a key
def encryptText(plaintext, key):
    ciphertext = ''
    for i in range(len(plaintext)):
        # Check if the character is lowercase to use the right ASCII codes
        if (plaintext[i].islower()):
            ciphertext += chr(97 + (ord(plaintext[i]) + key - 97) % 26)
        # Check if the character is uppercase to use the right ASCII codes
        elif (plaintext[i].isupper()):
            ciphertext += chr(65 + (ord(plaintext[i]) + key - 65) % 26)
        # If the character is not alphabetical just add it to the ciphertext
        else:
            ciphertext += plaintext[i]
    return ciphertext
    
    
if __name__ == "__main__":
    main()